### PR TITLE
net-nntp/sabnzbd: add 4.1.0, 4.2.1

### PR DESCRIPTION
**Pull request by joecool1029**

### CODE REVIEW

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+
+PYTHON_COMPAT=( python3_{9..11} )
```
Comment (dlan17) [inline net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:6]: can you update the PYTHON_COMPAT and re-test? pls check version 4.0.3

Comment (joecool1029) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:6]: That was deliberate. Officially upstream started supporting 3.12 in the 4.2.0 release cycle.

But I can test the older versions and see if they pass tests with 3.12....

Comment (joecool1029) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:6]: Update: I am unable to test 3.12 yet on my two AMD64 systems, there's too many packages still requiring 3.11. My minimal install with 3.12 on ARM64 is not compatible with all the test packages, so I cannot advertise that arch for this.

Someone else will need to test if they want 3.12 for the old releases. I will advertise 3.12 for 4.2.0 when it releases (should be soon, on RC2 now).

Comment (thesamesam) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:6]: I'm surprised you can't test it - you should be able to put */* PYTHON_TARGETS: python3_11 python3_12 in package.use?

Comment (joecool1029) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:6]: yeah I can do that, I forgot python supports multiple targets at once. Been building systems single target for years....

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+PYTHON_REQ_USE="sqlite"
+
+inherit optfeature python-single-r1 systemd
+
+MY_PV="${PV/_alpha/Alpha}"
+MY_PV="${MY_PV/_beta/Beta}"
+MY_PV="${MY_PV/_rc/RC}"
+
+MY_P="${PN/sab/SAB}-${MY_PV}"
```
Comment (mgorny) [inline net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:15]: Just inline the package name here, substitution is longer than doing that, it's unreadable and it's just asking for trouble.

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		dev-python/notify2[${PYTHON_USEDEP}]
+		dev-python/portend[${PYTHON_USEDEP}]
+		dev-python/puremagic[${PYTHON_USEDEP}]
+		~dev-python/sabctools-7.1.2[${PYTHON_USEDEP}]
+	')
+		test? ( $(python_gen_cond_dep '
+				dev-python/tavalidate[${PYTHON_USEDEP}]
+				>=dev-python/tavern-2[${PYTHON_USEDEP}]
+			')
+		)
```
Comment (mgorny) [inline net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:51]: Wrong indentation.

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+"
+RDEPEND="
+	${DEPEND}
```
Comment (mgorny) [inline net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:54]: This copies test dependencies into RDEPEND which is wrong.

Also, it seems that test dependencies are now randomly split between `DEPEND` and `BDEPEND`.

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	local -x PYTEST_PLUGINS=tavern._core.pytest
+	epytest -s
+}
+
+src_install() {
+	local d
+	for d in email icons interfaces locale po sabnzbd scripts tools; do
+		insinto /usr/share/${PN}/${d}
+		doins -r ${d}/*
+	done
```
Comment (mgorny) [inline net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:129]: Any reason not to do the equivalent of:

```
insinto /usr/share/${PN}
doins -r email icons ...
```

?

**Review by mgorny — CHANGES_REQUESTED** [review]

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild (hunk 1 above)
Comment (joecool1029) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild]: Sorry for delay, passes 3.12 tests!

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild (hunk 4 above)
Comment (joecool1029) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild:49]: Good catch, I've moved all the test dependencies to BDEPEND.

File: net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild (hunk 5 above)
Comment (joecool1029) [reply net-nntp/sabnzbd/sabnzbd-4.1.0.ebuild]: Didn't know I could, inherited from old EAPI where maybe was not possible. It's changed now, thanks!

**Comment by joecool1029** [comment]
I've tacked on the 4.2.0 release. Dependencies updated, I think the QA scan failed due to me initially not rebasing off the current tree (sabctools was bumped since then so it should pass now?)

**Comment by joecool1029** [comment]
@mgorny can you please approve?

**Review by juippis — APPROVED** [review]
I'll just grab the 4.2.1 here, 4.1.0 seems useless after that being merged? And can't drop 4.0.2 due to it having stable keywords.

**Pull request closed without merging**